### DISCUSSION TITLE
Tracking module version adoption

### DISCUSSION THREAD

**Discussion by pi0** [community/community#128450] [npm]
Hi! First thanks for all the awesome stuff you are doing to make npm happen! 

Recently, npm registry supports download stats by version for last 7-days. As stated in [blog post](https://github.blog/changelog/2021-01-27-npmjs-com-displays-per-version-download-counts/):

> This change will help developers understand the popularity of different versions of their packages, and which versions are still in common use.

This number, however, helps for understanding the popularity of each tag based on the number of downloads in **the last-week** but without a time-series of data, these numbers are misleading about **adoption status** of each version. Since adoption is defined during time.

I found an old related issue (npm/download-counts#6). If we can query version download stats for custom or specific ranges other than last 7 days, we can compute migration speed and real adoption like this:

![image](https://user-images.githubusercontent.com/5158436/113041897-cae61700-919a-11eb-9d73-9fd2aad1ddc4.png)

An alternative would be keeping a snapshot for packages from versions-api each week but I think it could be really useful information if npm registry could do it. 

---

Reference: Apis i found for query:

- https://api.npmjs.org/downloads/range/last-week/{pkg}
- https://api.npmjs.org/downloads/range/2020-09-27:2021-03-27/{pkg}
- https://api.npmjs.org/versions/{pkg}/last-week
- [versions/{pkg}/[a:b] does not exists

Related PR: https://github.com/npm/registry/pull/9

**Comment by MylesBorins** [comment]
hey @pi0 thanks for posting. I'm following up with the engineering team to see what the status quo is for this data and what we might be able to offer longer term
